File: java/src/main/java/io/vedadb/wire/vbp/VBPError.java
package io.vedadb.wire.vbp;

/** High-level VBP error decoded from an ERROR frame body. */
public class VBPError extends RuntimeException {
    public final String sqlstate;
    public final String detail;
    public final String hint;

    public VBPError(String sqlstate, String message) {
        this(sqlstate, message, "", "");
    }

    public VBPError(String sqlstate, String message, String detail, String hint) {
        super("[" + sqlstate + "] " + message);
        this.sqlstate = sqlstate;
        this.detail = detail;
        this.hint = hint;
    }
}

File: dotnet/VedaDB/Wire/Vbp/VBPError.cs
// VedaDB .NET SDK — VBP wire layer

using System;

namespace VedaDB.Wire.Vbp
{
    /// <summary>
    /// High-level VBP error decoded from an ERROR frame body.
    /// Carries SQLSTATE plus optional detail and hint.
    /// </summary>
    public class VBPErrorException : Exception
    {
        public string SqlState { get; }
        public string Detail { get; }
        public string Hint { get; }

        public VBPErrorException(string sqlState, string message)
            : this(sqlState, message, "", "")
        {
        }

        public VBPErrorException(string sqlState, string message, string detail, string hint)
            : base("[" + sqlState + "] " + message)
        {
            SqlState = sqlState;
            Detail = detail ?? "";
            Hint = hint ?? "";
        }
    }

    /// <summary>
    /// Wraps a SQLSTATE for callers that want typed exception access.
    /// </summary>
    public class VBPException : Exception
    {
        public string SqlState { get; }
        public VBPException(string sqlState, string message) : base("[" + sqlState + "] " + message)
        {
            SqlState = sqlState;
        }
    }
}
